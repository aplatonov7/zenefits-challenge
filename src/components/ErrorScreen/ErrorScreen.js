import React from 'react'
import PropTypes from 'prop-types'

import './ErrorScreen.css'

const ErrorScreen = ({ children, title }) => (
  <div className="ErrorScreen__container">
    <h1>{title || 'Sorry, something went terribly wrong'}</h1>
    {children}
  </div>
)

ErrorScreen.propTypes = {
  children: PropTypes.node,
  title: PropTypes.string,
}

export default ErrorScreen