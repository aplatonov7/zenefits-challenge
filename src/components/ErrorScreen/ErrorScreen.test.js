import React from 'react'

import ErrorScreen from './ErrorScreen'

describe('ErrorScreen component', () => {
  let component

  beforeEach(() => {
    component = shallow(<ErrorScreen />)
  })

  it('Should match snapshot', () => {
    const props = { children: 'Inner content', title: 'Custom title' }
    expect(renderer.create(<ErrorScreen {...props} />).toJSON()).toMatchSnapshot()
  })

  it('Should render title', () => {
    component.setProps({ title: 'Custom title' })
    expect(component.find('h1').text()).toEqual('Custom title')
  })

  it('Should render children', () => {
    component.setProps({ children: 'Inner content' })
    expect(component.contains('Inner content')).toBe(true)
  })
})