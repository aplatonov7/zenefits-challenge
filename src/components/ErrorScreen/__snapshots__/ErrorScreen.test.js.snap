// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorScreen component Should match snapshot 1`] = `
<div
  className="ErrorScreen__container"
>
  <h1>
    Custom title
  </h1>
  Inner content
</div>
`;
